hm_pyhelper/gateway_grpc/client.py

```diff
@@ -0,0 +1,231 @@
+import base58
+import grpc
+import subprocess
+import json
+
+from hm_pyhelper.protos import blockchain_txn_add_gateway_v1_pb2, \
+    local_pb2_grpc, local_pb2, region_pb2
+from hm_pyhelper.gateway_grpc.exceptions import GatewayMalformedAddGatewayTxn
+
+from hm_pyhelper.logger import get_logger
+
+LOGGER = get_logger(__name__)
+
+
+def decode_pub_key(encoded_key: bytes) -> str:
+    # Addresses returned by the RPC response are missing a leading
+    # byte for the version. The version is currently always 0.
+    # https://github.com/helium/helium-js/blob/8d5cb76e156fb80de6fc80f239b43e3872c7b7d7/packages/crypto/src/Address.ts#L64
+    version_byte = b'\x00'
+
+    # Convert binary address to base58
+    complete_key = version_byte + encoded_key
+    decoded_key = base58.b58encode_check(complete_key).decode()
+    return decoded_key
+
+
+class GatewayClient(object):
+    '''
+    GatewayClient wraps grpc api provided by helium gateway-rs
+    It provides some convenience methods to support the old api
+    to limit breaking changes.
+    Direct interaction with the grpc api can be achieved by
+    using GatewayClient.stub.<api>
+
+    All methods might return grpc pass through exceptions.
+    '''
+
+    def __init__(self, url='helium-miner:4467'):
+        self._url = url
+        self._channel = grpc.insecure_channel(url)
+        self._channel.subscribe(self._connect_state_handler)
+        self.stub = local_pb2_grpc.apiStub(self._channel)
+
+    def _connect_state_handler(self, state):
+        if state == grpc.ChannelConnectivity.SHUTDOWN:
+            LOGGER.error('GRPC Channel shutdown : irrecoverable error')
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, _, _2, _3):
+        self._channel.close()
+
+    def get_validator_info(self) -> local_pb2.height_res:
+        return self.stub.height(local_pb2.height_req())
+
+    def get_height(self) -> int:
+        return self.get_validator_info().height
+
+    def get_region_enum(self) -> int:
+        '''
```

Comment:
Letting possible grpc exception pass through. Caller might be in a better position.
Catching them here and raising MinerFetchException might be good abstraction but I didn't see any value in that.